src/PKSim.Core/Commands/RemoveMoleculeFromIndividualCommand.cs

```diff
@@ -29,8 +30,8 @@ protected override void ClearReferences()
 
       protected override void PerformExecuteWith(IExecutionContext context)
       {
-         _individual.RemoveMolecule(_molecule);
-         context.Unregister(_molecule);
+         var removedContainer = _individual.RemoveMolecule(_molecule);
```

Comment:
This was also a bug that I fixed at the same time (same logical error as the actual bug). Removing a molecule only unregistered the TOP container but not all sub containers.